yt_dlp/extractor/amcnetworks.py

```diff
@@ -65,15 +65,36 @@ class AMCNetworksIE(ThePlatformIE):
     def _real_extract(self, url):
         site, display_id = re.match(self._VALID_URL, url).groups()
         requestor_id = self._REQUESTOR_ID_MAP[site]
-        properties = self._download_json(
-            'https://content-delivery-gw.svc.ds.amcn.com/api/v2/content/amcn/%s/url/%s' % (requestor_id.lower(), display_id),
-            display_id)['data']['properties']
+        page_data = self._download_json(
+            'https://content-delivery-gw.svc.ds.amcn.com/api/v2/content/amcn/%s/url/%s'
+            % (requestor_id.lower(), display_id), display_id)['data']
+        properties = page_data.get('properties')
         query = {
             'mbr': 'true',
             'manifest': 'm3u',
         }
-        tp_path = 'M_UwQC/media/' + properties['videoPid']
-        media_url = 'https://link.theplatform.com/s/' + tp_path
+
+        has_releasePid = False
+        video_player_count = 0
+        try:
+            for v in page_data['children']:
+                if v['type'] == 'video-player':
+                    releasePid = \
+                        v['properties']['currentVideo']['meta']['releasePid']
+                    tp_path = 'M_UwQC/' + releasePid
+                    media_url = 'https://link.theplatform.com/s/' + tp_path
+                    has_releasePid = True
+                    video_player_count += 1
+        except KeyError:
+            pass
+        if video_player_count > 1:
+            self.report_warning('The JSON data has more than one video player.')
+        # This extractor originally used videoPid instead of releasePid
+        # but some of those resulting manifests have DRM.
+        if not has_releasePid:
+            tp_path = 'M_UwQC/media/' + properties['videoPid']
+            media_url = 'https://link.theplatform.com/s/' + tp_path
```

Comment:
```suggestion        media_url = None
        for v in (page_data.get('children') or {}):
            if not v.get('type') == 'video-player':
                continue
            releasePid = try_get(
                v, lambda x: x['properties']['currentVideo']['meta']['releasePid'])
            if not releasePid:
                continue
            if media_url:
                self.report_warning('The JSON data has more than one video player.')
                break
            tp_path = 'M_UwQC/' + releasePid
            media_url = 'https://link.theplatform.com/s/' + tp_path
        # Fallback to videoPid which may have DRM
        if not media_url:
            tp_path = 'M_UwQC/media/' + properties['videoPid']
            media_url = 'https://link.theplatform.com/s/' + tp_path
```

Untested. Removes unnecessary variables and try block

Reply:
Of course, the warning message could be changed to report the number of video-player values encountered.

Also, the fallback approach has a flaw: If the releasePid manifest exists and has DRM, the download will fail even if videoPid has a valid manifest. The program would need a function/method to test for DRM and possibly report what form of DRM was encountered. That'd be more informative than generic messages that no formats were found.

Reply:
> Also, the fallback approach has a flaw: If the releasePid manifest exists and has DRM, the download will fail even if videoPid has a valid manifest.

I dont know enough about this extractor/website to make judgement call on that. I simply refactored your existing code. If the `videoPid` could have formats that are not available through `releasePid`, you could extract formats from both and append them together

> The program would need a function/method to test for DRM and possibly report what form of DRM was encountered.

If the extractor is able to detect that the video is undownloadable due to DRM, it can `raise ExtractorError('This video is DRM protected.', expected=True)`. `_parse_smil_formats` cannot raise this error since the extractor may have more than one manifest, some of which could have non-DRM formats

Reply:
The way I wrote it, the entire try block fails as soon as one key isn't found. The way you wrote it has to deal with more conditions; also, it has one more loop (within try_get).

Having more than one "video-player" JSON value doesn't mean that the page necessarily has more than one video. That would depend on how a page actually uses the JSON info. A page could be implemented, for example, to use only the first video-player entry or the last, and have the data for the others not include a releasePid value.

Reply:
> The way I wrote it, the entire try block fails as soon as one key isn't found. The way you wrote it has to deal with more conditions; also, it has one more loop (within try_get).

Yes, that is the intention. In your code, if just one of the `children` has no `type` or `releasePid`, the whole loop will fail. Was that your intention? if so, please explain why

> A page could be implemented, for example, to use only the first video-player entry or the last, and have the data for the others not include a releasePid value.

The warning is issued only when a second releasePid is found

Reply:
>  `_parse_smil_formats` cannot raise this error since the extractor may have more than one manifest, some of which could have non-DRM formats

Yes, apparently, each of these videos has at least two different master manifests.

> In your code, if just one of the children has no type or releasePid, the whole loop will fail. Was that your intention? if so, please explain why

Yes, the loop should fail. The video pages have only one key/value pair for 'type':'video-player'. If that lacks a releasePid key, then either the page uses videoPid to specify the video (the fallback) or it's doing something entirely new. The valid URLs for this extractor are web pages for individual episodes of a TV show; they shouldn't have more than one video per page.

Reply:
You can use break instead of continue then, no?
```py
            if not releasePid:
                break
```

Reply:
```py
if v['type'] == 'video-player'
```

This will cause the `KeyError` to trigger if any of the children doesnt have a `type`.

Reply:
If any of the children lacked `type` then the video's web page would have problems but I suppose yt-dlp doesn't have to rely on the page displaying properly.

Reply:
What's the issue with using try/except? Also, in this case, isn't using try_get less efficient? It also uses try/except but additionally involves passing parameters and using an extra `for` loop.

Reply:
> isn't using try_get less efficient? It also uses try/except but additionally involves passing parameters and using an extra for loop.

Such minor performance differences does not matter!

Extractors generally dont use `try/except` except to catch `ExtractorError` from some common function. Hence why I suggested not to use it. But looking through the code, there are a few extractors that do use it. You can keep it as-is if you want to so badly. Just make sure the error is thrown only when needed. Eg: `v['type']`